### PR TITLE
Lpp 40453 commerce 5908

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://issues.liferay.com/browse/COMMERCE-5908

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPSpecificationOptionServiceImpl.java [hunk 1]
```diff
@@ -95,6 +95,23 @@ public CPSpecificationOption getCPSpecificationOption(
...
+		throws PortalException {
+
+		CPSpecificationOption cpSpecificationOption =
+			cpSpecificationOptionLocalService.getCPSpecificationOption(
+				companyId, key);
+
+		if (cpSpecificationOption != null) {
+			_cpSpecificationOptionModelResourcePermission.check(
+				getPermissionChecker(), cpSpecificationOption, ActionKeys.VIEW);
+		}
```
Comment (dsitu) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPSpecificationOptionServiceImpl.java:110]: @andrea-ale-sbarra I don't think we need the null check here. Wont `cpSpecificationOptionLocalService.getCPSpecificationOption(companyId, key);`
just throw `NoSuchCPSpecificationOptionException` if it does not find a cpSpecificaionOption?

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPSpecificationOptionServiceImpl.java]: you're totally right! thanks for the catch @dsitu

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:stop

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**